new-lamassu-admin/src/pages/AddMachine/AddMachine.js

```diff
@@ -155,45 +184,51 @@ const renderStepper = (step, it, idx, classes) => {
   )
 }
 
-const AddMachine = memo(({ close }) => {
+const AddMachine = memo(({ close, onPaired }) => {
   const classes = useStyles()
-  const [qrCode, setQrCode] = useState(null)
-  const [step, setStep] = useState(0)
+  const { data } = useQuery(GET_MACHINES)
+  const [payload, setPayload] = useSetState({ qrcode: '', name: '', count: 0 })
+  const [step, { inc }] = useCounter(0, steps.length, 0)
+  const count = getSize(data)
+
+  useEffect(() => setPayload({ count }), [count, setPayload])
```

Comment:
We don't really need count in the payload since internal components don't need to change it. Simply pass down count as a prop to `<Component>`